test/Microsoft.IdentityModel.Tokens.Saml.Tests/SamlSerializerTests.cs

```diff
@@ -0,0 +1,308 @@
+            var context = new CompareContext($"{this}.ReadAction, {theoryData.TestId}");
+            try
+            {
+                var sr = new StringReader(theoryData.ActionTestSet.Xml);
+                var reader = XmlDictionaryReader.CreateDictionaryReader(XmlReader.Create(sr));
+                var action = (theoryData.SamlSerializer as SamlSerializerPublic).ReadActionPublic(reader);
+                theoryData.ExpectedException.ProcessNoException();
+
+                IdentityComparer.AreEqual(action, theoryData.ActionTestSet.Action, context);
+            }
+            catch (Exception ex)
+            {
+                theoryData.ExpectedException.ProcessException(ex);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        public static TheoryData<SamlTheoryData> ReadActionTheoryData
+        {
+            get
+            {
+                return new TheoryData<SamlTheoryData>
+                {
+                    new SamlTheoryData
+                    {
+                        ExpectedException = new ExpectedException(typeof(SamlSecurityTokenReadException), "IDX11112:", typeof(ArgumentNullException)),
+                        First = true,
+                        SamlSerializer = new SamlSerializerPublic(),
+                        ActionTestSet = ReferenceXml.SamlActionValueNull,
+                        TestId = nameof(ReferenceXml.SamlActionValueNull)
+                    },
+                    new SamlTheoryData
+                    {
+                        ExpectedException = new ExpectedException(typeof(SamlSecurityTokenReadException), "IDX11112:", typeof(ArgumentNullException)),
+                        First = true,
+                        SamlSerializer = new SamlSerializerPublic(),
+                        ActionTestSet = ReferenceXml.SamlActionValueEmpthString,
+                        TestId = nameof(ReferenceXml.SamlActionValueEmpthString)
+                    },
+                    new SamlTheoryData
+                    {
+                        ExpectedException = ExpectedException.NoExceptionExpected,
+                        SamlSerializer = new SamlSerializerPublic(),
+                        ActionTestSet = ReferenceXml.SamlActionNoNamespace,
+                        TestId = nameof(ReferenceXml.SamlActionNoNamespace)
+                    },
+                    new SamlTheoryData
+                    {
+                        ExpectedException = new ExpectedException(typeof(SamlSecurityTokenReadException), "IDX11111:"),
+                        SamlSerializer = new SamlSerializerPublic(),
+                        ActionTestSet = ReferenceXml.SamlActionInvalidNamespace,
+                        TestId = nameof(ReferenceXml.SamlActionInvalidNamespace)
+                    },
+                    new SamlTheoryData
+                    {
+                        ExpectedException = ExpectedException.NoExceptionExpected,
```

Comment:
@ChenYan98 have you considered using the Thumb Up reaction to mark comments as fixed? Otherwise your subscribers get an e-mail every time you say Fixed.

@brentschmaltz The review feature would also let you batch comments to avoid a similar flood.